src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystem.java

```diff
@@ -350,6 +373,70 @@ public static EquationSystem create(LfNetwork network, VariableSet variableSet,
         createBusEquations(network, variableSet, creationParameters, equationSystem);
         createBranchEquations(network, variableSet, creationParameters, equationSystem);
 
+        network.addListener(new LfNetworkListener() {
+            @Override
+            public void onVoltageControlChange(LfBus bus, boolean oldVoltageControl, boolean newVoltageControl) {
+                if (newVoltageControl) { // switch PQ/PV
+                    Equation qEq = equationSystem.createEquation(bus.getNum(), EquationType.BUS_Q);
+                    qEq.setActive(false);
+
+                    LfBus controlledBus = bus.getControlledBus().orElse(null);
+                    if (controlledBus != null) {
+                        updateControlledBus(controlledBus, equationSystem, variableSet);
+                    } else {
+                        Equation vEq = equationSystem.createEquation(bus.getNum(), EquationType.BUS_V);
+                        vEq.setActive(true);
+                    }
+                } else { // switch PV/PQ
+                    Equation qEq = equationSystem.createEquation(bus.getNum(), EquationType.BUS_Q);
+                    qEq.setActive(true);
+
+                    LfBus controlledBus = bus.getControlledBus().orElse(null);
+                    if (controlledBus != null) {
+                        updateControlledBus(controlledBus, equationSystem, variableSet);
+                    } else {
+                        Equation vEq = equationSystem.createEquation(bus.getNum(), EquationType.BUS_V);
+                        vEq.setActive(false);
+                    }
+                }
+            }
+
+            @Override
+            public void onPhaseControlModeChange(DiscretePhaseControl phaseControl, DiscretePhaseControl.Mode oldMode, DiscretePhaseControl.Mode newMode) {
+                boolean on = newMode != DiscretePhaseControl.Mode.OFF;
```

Comment:
Here, the code on master is: 
```
LfBranch controllerBranch = branch.getDiscretePhaseControl().getController();
 Variable a1 = context.getVariableSet().getVariable(controllerBranch.getNum(), VariableType.BRANCH_ALPHA1);
 a1.setActive(false);

 Equation t = context.getEquationSystem().createEquation(branch.getNum(), EquationType.BRANCH_P);
 t.setActive(false);
```
Why we don't have any variable management?

Reply:
I reverted the phase control refactoring. I will add it in another PR and will explain better why.